components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/processors/RandomPasswordProcessor.java

```diff
@@ -132,7 +132,7 @@ private void addPasswordContainerToCache(RandomPassword[] randomPasswords, Strin
         RandomPasswordContainerCache.getInstance().addToCache(cacheKey, cacheEntry);
     }
 
-    private RandomPassword[] getAndRemoveRandomPasswordContainerFromCache(String uuid) {
+    private RandomPassword[] getAndRemoveRandomPasswordContainerFromCache(String uuid, boolean withCacheClear) {
```

Comment:
method name should be changed as with the new parameter its not always clearing the entry